SpiffWorkflow/SpiffWorkflow/bpmn/specs/events/event_definitions.py

```diff
@@ -72,9 +73,9 @@ def _throw(self, event, workflow, outer_workflow, correlations=None):
         # We also don't have a more sophisticated method for addressing events to
         # a particular process, but this at least provides a mechanism for distinguishing
         # between processes and subprocesses.
-        if self.external:
```

Comment:
This code is pile of crap so if this change doesn't cause any issues, you might as well change it.  The new code isn't going to be any worse what's there.  Certain events are only applicable within the workflow where they originated and others aren't handled by their originating workflow and that's what this code is trying (badly) to handle.

The historical context of this is transactions (cancel events generated inside subprocesses are supposed to be sent to the outer workflow).  But sometimes I look at this code and say to myself "what the hell was I thinking?".  So no objections to changing this if necessary.

I think there could be problems at some point, but I expect they would be the result of underlying issues rather than these specific changes.